### PR TITLE
`ultralytics 8.3.25` Alibaba MNN export and predict support

### PR DESCRIPTION
**Pull request by wangzhaode**
Support Alibaba [MNN](https://github.com/alibaba/MNN) export and predict.

Test code:
```python
from ultralytics import YOLO

model = YOLO("yolo11n.pt")

model.export(format='mnn')

mnn_model = YOLO("yolo11n.mnn")
results = mnn_model('../resource/test.jpg')

for result in results:
    result.show()  # display to screen
    result.save(filename="result.jpg")  # save to disk
```

image

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://github.com/ultralytics/actions)<sub>

### 🌟 Summary
This PR introduces support for exporting and running YOLO11 models using the new MNN (Mobile Neural Network) format, catering to efficient deployment on mobile and embedded devices.

### 📊 Key Changes
- **.gitignore Update**: Added `*.mnn` to ensure MNN model files are ignored.
- **Documentation Enhancements**: Updated deployment guides and integration sections to include MNN.
- **New MNN Documentation**: Added detailed docs for using MNN with YOLO11 models, including setup, export, and inference examples.
- **MNN Integration**: Added the MNN export option in the codebase, allowing models to be exported in MNN format.
- **Testing**: Introduced a test case for exporting models to the MNN format.
- **Version Bump**: Updated version number from `8.3.24` to `8.3.25`.

### 🎯 Purpose & Impact
- **Purpose**: To provide users with the ability to deploy YOLO11 models efficiently on mobile and embedded devices using the MNN format, which excels in performance for on-device inference.
- **Impact**: This broadens the usability of YOLO models on diverse platforms such as IoT, enhancing the model's reach in real-time mobile applications and embedded systems. The new format supports both Python and C++ deployments, catering to wider developer needs.

Overall, this update makes YOLO11 models more versatile across different hardware platforms, particularly optimizing for those with limited resources. 🚀📱

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @wangzhaode, thank you for submitting an Ultralytics 🚀 pull request! This is an automated response to assist you in the PR process. An Ultralytics engineer will review your submission soon. Meanwhile, please ensure the following checklist is complete:

- ✅ **Define a Purpose**: Clearly explain the purpose of your update, linking to any [relevant issues](https://github.com/ultralytics/ultralytics/issues). Ensure your commit messages are clear and follow project conventions.
- ✅ **Synchronize with Source**: Make sure your branch is up-to-date with the `ultralytics/ultralytics` `main` branch. If needed, update it by clicking 'Update branch' or running `git pull` and `git merge main` locally.
- ✅ **Ensure CI Checks Pass**: Confirm that all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI/) checks are passing. Address any issues if checks fail.
- ✅ **Update Documentation**: Check and update the relevant [documentation](https://docs.ultralytics.com) for any new or modified features.
- ✅ **Add Tests**: If applicable, include or update tests to cover your changes, ensuring all tests pass.
- ✅ **Sign the CLA**: If this is your first PR, make sure you've signed the [Contributor License Agreement](https://docs.ultralytics.com/help/CLA/) by commenting "I have read the CLA Document and I sign the CLA."
- ✅ **Minimize Changes**: Limit changes to the **minimum** required for your feature or fix. _"It is not daily increase but daily decrease, hack away the unessential."_

For further guidance, please review our [Contributing Guide](https://docs.ultralytics.com/help/contributing). If you need assistance, feel free to add a comment here. Thank you for enhancing Ultralytics with Alibaba MNN support! 🌟💻🚀

**Comment by wangzhaode** [comment]
I have read the CLA Document and I sign the CLA

**Comment by wangzhaode** [comment]
recheck

**Review by Blingz14 — APPROVED** [review]

**Review by Blingz14 — APPROVED** [review]

**Review by Blingz14 — APPROVED** [review]

**Review by Blingz14 — APPROVED** [review]

**Review by Blingz14 — APPROVED** [review]

**Review by Blingz14 — APPROVED** [review]

**Review by Blingz14 — APPROVED** [review]

**Review by Blingz14 — APPROVED** [review]

**Review by ambitious-octopus — COMMENTED** [review]
Hi @wangzhaode, thanks for the PR. It's a great idea and great work! I just tested the PR, and the performances seem interesting. This export is only for half-precision CPU, right? In the official documentation, I read that it also supports `int8` precision export and runtime on different devices. Do you think it's too early to integrate them?
There is still some polishing work to do in this PR, so I'll do a more in-depth review.

**Comment by glenn-jocher** [comment]
@wangzhaode please ensure all CI checks are passing and that your branch is up-to-date with the main branch. Let us know if you need further assistance!

**Comment by inisis** [comment]
Great job, MNN is awesome.

**Comment by glenn-jocher** [comment]
Thank you! The credit goes to the YOLO community and the Ultralytics team for their continuous efforts. If you have any questions or need assistance, feel free to ask!

**Comment by wangzhaode** [comment]
> Hi @wangzhaode, thanks for the PR. It's a great idea and great work! I just tested the PR, and the performances seem interesting. This export is only for half-precision CPU, right? In the official documentation, I read that it also supports `int8` precision export and runtime on different devices. Do you think it's too early to integrate them? There is still some polishing work to do in this PR, so I'll do a more in-depth review.

Thank you for your response. The current PR now supports exporting and running in fp16 and int8 formats, and I've also added documentation to explain these features.

**Comment by glenn-jocher** [comment]
@wangzhaode thank you for the update. It's great to hear that fp16 and int8 support has been added. Looking forward to your detailed review.

File: ultralytics/engine/exporter.py [hunk 1]
```diff
@@ -604,6 +611,28 @@ def export_ncnn(self, prefix=colorstr("NCNN:")):
         yaml_save(f / "metadata.yaml", self.metadata)  # add metadata.yaml
         return str(f), None
 
+    @try_export
+    def export_mnn(self, prefix=colorstr("MNN:")):
+        """YOLOv8 MNN export using MNN https://github.com/alibaba/MNN."""
+        check_requirements("MNN")
+        import MNN  # noqa
```
Comment (inisis) [inline ultralytics/engine/exporter.py:618]: curious to ask, why use capital MNN as as package name

Comment (wangzhaode) [reply ultralytics/engine/exporter.py]: The project name is [MNN](https://github.com/alibaba/MNN), just using it.

**Comment by wangzhaode** [comment]
All CI checks are passing and the branch is up-to-date with the main branch.

File: ultralytics/engine/exporter.py [hunk 2]
```diff
@@ -4,19 +4,20 @@
...
-TensorRT                | `engine`                  | yolo11n.engine
-CoreML                  | `coreml`                  | yolo11n.mlpackage
-TensorFlow SavedModel   | `saved_model`             | yolo11n_saved_model/
-TensorFlow GraphDef     | `pb`                      | yolo11n.pb
-TensorFlow Lite         | `tflite`                  | yolo11n.tflite
-TensorFlow Edge TPU     | `edgetpu`                 | yolo11n_edgetpu.tflite
-TensorFlow.js           | `tfjs`                    | yolo11n_web_model/
-PaddlePaddle            | `paddle`                  | yolo11n_paddle_model/
-NCNN                    | `ncnn`                    | yolo11n_ncnn_model/
+PyTorch                 | -                         | yolov8n.pt
```
Comment (Laughing-q) [inline ultralytics/engine/exporter.py:7]: `yolov8n` --> `yolo11n` :)

File: ultralytics/engine/exporter.py [hunk 3]
```diff
@@ -30,18 +31,19 @@
...
-                         yolo11n_openvino_model     # OpenVINO
-                         yolo11n.engine             # TensorRT
-                         yolo11n.mlpackage          # CoreML (macOS-only)
-                         yolo11n_saved_model        # TensorFlow SavedModel
-                         yolo11n.pb                 # TensorFlow GraphDef
-                         yolo11n.tflite             # TensorFlow Lite
-                         yolo11n_edgetpu.tflite     # TensorFlow Edge TPU
-                         yolo11n_paddle_model       # PaddlePaddle
-                         yolo11n_ncnn_model         # NCNN
+    $ yolo predict model=yolov8n.pt                 # PyTorch
```
Comment (Laughing-q) [inline ultralytics/engine/exporter.py:34]: the same as above

**Review by inisis — APPROVED** [review]
LGTM, impressive work

File: ultralytics/engine/exporter.py (hunk 3 above)
Comment (wangzhaode) [reply ultralytics/engine/exporter.py]: fixed

File: ultralytics/engine/exporter.py (hunk 2 above)
Comment (wangzhaode) [reply ultralytics/engine/exporter.py]: fixed

**Comment by Laughing-q** [comment]
@wangzhaode Thanks for the PR! It looks amazing! One question though, can we put the exported model and metadata.yaml in to a folder? i.e. `yolo11n_mnn/` so users can easily find and load the model and `metadata.yaml` then do inference. Just like we did for ncnn/openvino:
pic-241014-1506-18
Thanks!

**Comment by wangzhaode** [comment]
> @wangzhaode Thanks for the PR! It looks amazing! One question though, can we put the exported model and metadata.yaml in to a folder? i.e. `yolo11n_mnn/` so users can easily find and load the model and `metadata.yaml` then do inference. Just like we did for ncnn/openvino: pic-241014-1506-18 Thanks!



> metadata.yaml

Thank you for your reply. The MNN model supports metadata storage, but the Python interface is not yet perfect for reading metadata. Therefore, after the next version of MNN (I have added support for metadata reading), I will provide a patch to fix the metadata storage method here. `metadata.yaml` is no longer used, so the folder format is not used here. I don't know if it is appropriate to temporarily use an independent `metadata.yaml` file? If not, I can change it to a folder format.

**Comment by wangzhaode** [comment]
> @wangzhaode Sounds good! can we wait for the patch released and directly remove the `metadata.yaml` in the PR? then I think everything is ready to go. :)



> metadata.yaml

Yes, I will modify this PR and remove metadata.yaml after MNN 2.9.6 is released at the end of the month.

**Comment by wangzhaode** [comment]
> @wangzhaode Sounds good! can we wait for the patch released and directly remove the `metadata.yaml` in the PR? then I think everything is ready to go. :)

Yes, I will modify this PR and remove metadata.yaml after the recent release of MNN 2.9.6. This PR can wait for a while.

**Comment by glenn-jocher** [comment]
Great, thank you for the update. We'll proceed once the MNN 2.9.6 release is available.

**Comment by wangzhaode** [comment]
MNN 2.9.6 was released today. Based on MNN 2.9.6, metadata.yml was deleted. Please review my PR again.

@Laughing-q @glenn-jocher

File: docs/en/guides/model-deployment-options.md [hunk 4]
```diff
@@ -258,6 +258,10 @@ NCNN is a high-performance neural network inference framework optimized for the
 
 - **Hardware Acceleration**: Tailored for ARM CPUs and GPUs, with specific optimizations for these architectures.
 
+#### MNN
+
+MNN is a highly efficient and lightweight deep learning framework. It supports inference and training of deep learning models and has industry-leading performance for inference and training on-device. In addition, MNN is also used on embedded devices, such as IoT.
+
 ## Comparative Analysis of YOLO11 Deployment Options
```
Comment (Laughing-q) [inline docs/en/guides/model-deployment-options.md:265]: @wangzhaode Can we have the MNN in the following table as well?

**Comment by Laughing-q** [comment]
@wangzhaode I noticed there's a hidden file generated when I export mnn format, do you know what's this?
```bash
yolo export model=yolo11n.pt format=mnn
```
oU9A4ciz0T

File: docs/en/guides/model-deployment-options.md (hunk 4 above)
Comment (Laughing-q) [reply docs/en/guides/model-deployment-options.md:265]: zHqqLhpo1I

**Comment by wangzhaode** [comment]
> @wangzhaode I noticed there's a hidden file generated when I export mnn format, do you know what's this?
> 
> ```shell
> yolo export model=yolo11n.pt format=mnn
> ```
> 
> oU9A4ciz0T

This is a temporary file used during the optimization pass.

File: docs/en/guides/model-deployment-options.md (hunk 4 above)
Comment (wangzhaode) [reply docs/en/guides/model-deployment-options.md:265]: Yeah, added in [edf0a77](https://github.com/ultralytics/ultralytics/pull/16802/commits/edf0a77d718ac857b7af84a2d14e442bbd416b17)

**Review by ambitious-octopus — CHANGES_REQUESTED** [review]
Hey @wangzhaode, I found a problem is that `int8` does not support cuda inference. Can we add a warning for users? Thanks!

**Review by ambitious-octopus — COMMENTED** [review]
@wangzhaode I updated our export table by adding mnn.

**Comment by glenn-jocher** [comment]
This is a temporary file created during the optimization process.

**Comment by glenn-jocher** [comment]
The hidden file is a temporary one created during the optimization process. It should be removed automatically after export.

**Comment by wangzhaode** [comment]
remove convert scratch file in [e671fde](https://github.com/ultralytics/ultralytics/pull/16802/commits/e671fde61ccb655cbe17fda012b767dc0078da39)

@Laughing-q @glenn-jocher

**Comment by glenn-jocher** [comment]
This is a temporary file created during the optimization process.

**Comment by Laughing-q** [comment]
> Hey @wangzhaode, I found a problem is that `int8` does not support cuda inference. Can we add a warning for users? Thanks!

@wangzhaode I tested the int8 inference with explicitly set `device=0` and it's running smoothly without any error. The Gpu is not used when I check `nvidia-smi` though, and the speed is still fast. I think it's all good at this stage.

**Review by Laughing-q — APPROVED** [review]

**Comment by glenn-jocher** [comment]
The hidden file is a temporary file used during the optimization process.

**Comment by Laughing-q** [comment]
@glenn-jocher This one is ready to go! The export the inference and validation all are good, and the MNN models performance is pretty good as well. If there's any undiscovered issues we could fix it afterwards. :)

**Comment by inisis** [comment]
> @glenn-jocher This one is ready to go! The export the inference and validation all are good, and the MNN models performance is pretty good as well. If there's any undiscovered issues we could fix it afterwards. :)

Exactly！

**Comment by glenn-jocher** [comment]
@wangzhaode the hidden file is a temporary file created during the optimization process.

**Comment by glenn-jocher** [comment]
@inisis yes this one is in the queue for tomorrow! We've been getting a lot of PRs and trying to launch patch updates with major changes like this one, so 8.3.25 should lead with this PR tomorrow.

File: docs/en/macros/export-table.md [hunk 5]
```diff
@@ -13,3 +13,4 @@
 | [TF.js](../integrations/tfjs.md)                  | `tfjs`            | `{{ model_name or "yolo11n" }}_web_model/`      | ✅       | `imgsz`, `half`, `int8`, `batch`                                     |
 | [PaddlePaddle](../integrations/paddlepaddle.md)   | `paddle`          | `{{ model_name or "yolo11n" }}_paddle_model/`   | ✅       | `imgsz`, `batch`                                                     |
 | [NCNN](../integrations/ncnn.md)                   | `ncnn`            | `{{ model_name or "yolo11n" }}_ncnn_model/`     | ✅       | `imgsz`, `half`, `batch`                                             |
+| [MNN](../integrations/mnn.md)                     | `mnn`             | `{{ model_name or "yolo11n" }}.mnn/`            | ✅       | `imgsz`, `batch`, `int8`, `half`                                     |
```
Comment (glenn-jocher) [inline docs/en/macros/export-table.md:16]: @wangzhaode if MNN is not a directory but a file then this last backslash should not be here.

**Comment by inisis** [comment]
@Laughing-q MNN should be placed ahead of NCNN, becasue after NCNN test, onnx export will fail, root cause is undefined

**Comment by Laughing-q** [comment]
@glenn-jocher @inisis opened a PR to test the case #17258

**Comment by Laughing-q** [comment]
@glenn-jocher @inisis @wangzhaode ok guys I've reverted the MNN/NCNN order changes I made today since I can't really find a quick solution to fix the crashing when exporting ONNX right after NCNN on windows. I don't have a windows machine so I tried to open a PR to debug it but can't really locate the error immediately this way.

**Comment by glenn-jocher** [comment]
@Laughing-q ah ok I understand. Then we should just update the docs tables to match the code order.

**Comment by inisis** [comment]
@Laughing-q cool

**Comment by Laughing-q** [comment]
@glenn-jocher yes updated!

**Comment by glenn-jocher** [comment]
@Laughing-q ok. Merging disable Ray test PR first and then this one before 15h here!

**Comment by Laughing-q** [comment]
> @glenn-jocher @inisis @wangzhaode ok guys I've reverted the MNN/NCNN order changes I made today since I can't really find a quick solution to fix the crashing when exporting ONNX right after NCNN on windows. I don't have a windows machine so I tried to open a PR to debug it but can't really locate the error immediately this way.

@glenn-jocher @inisis I asked @Burhan-Q to do a test for me on Windows but it's all good on his machine...really strange

**Comment by Burhan-Q** [comment]
For completeness, here's the output from my testing

<details><summary>Test result</summary>
<p>

```
Python 3.10.9 (tags/v3.10.9:1dd9be6, Dec  6 2022, 20:01:21) [MSC v.1934 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.

>>> from ultralytics import YOLO
>>> model = YOLO("yolo11n.pt")
>>> model.export(imgsz=160, format="ncnn", half=False, int8=False, device="cpu", verbose=False)

Ultralytics 8.3.18 🚀 Python-3.10.9 torch-2.3.1+cu121 CPU (Intel Core(TM) i5-10600K 4.10GHz)
YOLO11n summary (fused): 238 layers, 2,616,248 parameters, 0 gradients, 6.5 GFLOPs

PyTorch: starting from 'Q:\ML_dev\weights\yolo11n.pt' with input shape (1, 3, 160, 160) BCHW and output shape(s) (1, 84, 525) (5.4 MB)

TorchScript: starting export with torch 2.3.1+cu121...
TorchScript: export success ✅ 1.9s, saved as 'Q:\ML_dev\weights\yolo11n.torchscript' (10.4 MB)

NCNN: starting export with NCNN 1.0.20240410...
NCNN: running 'pnnx.exe Q:\ML_dev\weights\yolo11n.torchscript ncnnparam=Q:\ML_dev\weights\yolo11n_ncnn_model\model.ncnn.param ncnnbin=Q:\ML_dev\weights\yolo11n_ncnn_model\model.ncnn.bin ncnnpy=Q:\ML_dev\weights\yolo11n_ncnn_model\model_ncnn.py pnnxparam=Q:\ML_dev\weights\yolo11n_ncnn_model\model.pnnx.param pnnxbin=Q:\ML_dev\weights\yolo11n_ncnn_model\model.pnnx.bin pnnxpy=Q:\ML_dev\weights\yolo11n_ncnn_model\model_pnnx.py pnnxonnx=Q:\ML_dev\weights\yolo11n_ncnn_model\model.pnnx.onnx fp16=0 device=cpu inputshape="[1, 3, 160, 160]"'
pnnxparam = Q:\ML_dev\weights\yolo11n_ncnn_model\model.pnnx.param
pnnxbin = Q:\ML_dev\weights\yolo11n_ncnn_model\model.pnnx.bin
pnnxpy = Q:\ML_dev\weights\yolo11n_ncnn_model\model_pnnx.py
pnnxonnx = Q:\ML_dev\weights\yolo11n_ncnn_model\model.pnnx.onnx
ncnnparam = Q:\ML_dev\weights\yolo11n_ncnn_model\model.ncnn.param
ncnnbin = Q:\ML_dev\weights\yolo11n_ncnn_model\model.ncnn.bin
ncnnpy = Q:\ML_dev\weights\yolo11n_ncnn_model\model_ncnn.py
fp16 = 0
optlevel = 2
device = cpu
inputshape = [1,3,160,160]f32
moduleop =
############# pass_level0
inline module = torch.nn.modules.linear.Identity
inline module = ultralytics.nn.modules.block.Attention
inline module = ultralytics.nn.modules.block.Bottleneck
inline module = ultralytics.nn.modules.block.C2PSA
inline module = ultralytics.nn.modules.block.C3k
inline module = ultralytics.nn.modules.block.C3k2
inline module = ultralytics.nn.modules.block.DFL
inline module = ultralytics.nn.modules.block.PSABlock
inline module = ultralytics.nn.modules.block.SPPF
inline module = ultralytics.nn.modules.conv.Concat
inline module = ultralytics.nn.modules.conv.Conv
inline module = ultralytics.nn.modules.conv.DWConv
inline module = ultralytics.nn.modules.head.Detect
inline module = torch.nn.modules.linear.Identity
inline module = ultralytics.nn.modules.block.Attention
inline module = ultralytics.nn.modules.block.C2PSA
inline module = ultralytics.nn.modules.block.C3k
inline module = ultralytics.nn.modules.block.C3k2
inline module = ultralytics.nn.modules.block.DFL
inline module = ultralytics.nn.modules.block.PSABlock
inline module = ultralytics.nn.modules.block.SPPF
inline module = ultralytics.nn.modules.conv.Concat
inline module = ultralytics.nn.modules.conv.Conv
inline module = ultralytics.nn.modules.conv.DWConv
inline module = ultralytics.nn.modules.head.Detect

----------------

############# pass_level1
############# pass_level2
############# pass_level3
############# pass_level4
############# pass_level5
############# pass_ncnn
NCNN: export success ✅ 1.4s, saved as 'Q:\ML_dev\weights\yolo11n_ncnn_model' (10.0 MB)

Export complete (3.9s)
Results saved to Q:\ML_dev\weights
Predict:         yolo predict task=detect model=Q:\ML_dev\weights\yolo11n_ncnn_model imgsz=160
Validate:        yolo val task=detect model=Q:\ML_dev\weights\yolo11n_ncnn_model imgsz=160 data=/usr/src/ultralytics/ultralytics/cfg/datasets/coco.yaml
Visualize:       https://netron.app
'Q:\\ML_dev\\weights\\yolo11n_ncnn_model'
>>> model = YOLO("yolo11n.pt")
>>> model.export(imgsz=160, format="onnx", half=False, int8=False, device="cpu", verbose=False)
Ultralytics 8.3.18 🚀 Python-3.10.9 torch-2.3.1+cu121 CPU (Intel Core(TM) i5-10600K 4.10GHz)
YOLO11n summary (fused): 238 layers, 2,616,248 parameters, 0 gradients, 6.5 GFLOPs

PyTorch: starting from 'Q:\ML_dev\weights\yolo11n.pt' with input shape (1, 3, 160, 160) BCHW and output shape(s) (1, 84, 525) (5.4 MB)

ONNX: starting export with onnx 1.15.0 opset 17...
ONNX: slimming with onnxslim 0.1.34...
ONNX: export success ✅ 1.8s, saved as 'Q:\ML_dev\weights\yolo11n.onnx' (10.1 MB)

Export complete (1.9s)
Results saved to Q:\ML_dev\weights
Predict:         yolo predict task=detect model=Q:\ML_dev\weights\yolo11n.onnx imgsz=160
Validate:        yolo val task=detect model=Q:\ML_dev\weights\yolo11n.onnx imgsz=160 data=/usr/src/ultralytics/ultralytics/cfg/datasets/coco.yaml
Visualize:       https://netron.app
'Q:\\ML_dev\\weights\\yolo11n.onnx'
```

</p>
</details>

<details><summary>System info</summary>
<p>

```
Ultralytics 8.3.18 🚀 Python-3.10.9 torch-2.3.1+cu121 CUDA:0 (NVIDIA GeForce RTX 3080, 12288MiB)
Setup complete ✅ (12 CPUs, 31.9 GB RAM, 788.6/1863.0 GB disk)

OS                  Windows-10-10.0.19045-SP0
Environment         Windows
Python              3.10.9
Install             git
RAM                 31.86 GB
Disk                788.6/1863.0 GB
CPU                 Intel Core(TM) i5-10600K 4.10GHz
CPU count           12
GPU                 NVIDIA GeForce RTX 3080, 12288MiB
GPU count           1
CUDA                12.1

matplotlib          ✅ 3.8.1>=3.3.0
opencv-python       ✅ 4.8.1.78>=4.6.0
pillow              ✅ 9.3.0>=7.1.2
pyyaml              ✅ 6.0.1>=5.3.1
requests            ✅ 2.31.0>=2.23.0
scipy               ✅ 1.11.3>=1.4.1
torch               ✅ 2.3.1+cu121>=1.8.0
torchvision         ✅ 0.18.1+cu121>=0.9.0
tqdm                ✅ 4.66.1>=4.64.0
psutil              ✅ 5.9.6
py-cpuinfo          ✅ 9.0.0
thop                ✅ 0.1.1-2209072238>=0.1.1
pandas              ✅ 2.1.3>=1.1.4
seaborn             ✅ 0.13.0>=0.11.0

ncnn==1.0.20240410
onnx==1.15.0
```

</p>
</details>

**Comment by glenn-jocher** [comment]
@wangzhaode @Laughing-q @Burhan-Q PR merged, great work guys!!@

**Comment by glenn-jocher** [comment]
@wangzhaode MNN tests are failing on Windows and Raspberry Pi: https://github.com/ultralytics/ultralytics/actions/runs/11608221105

**Comment by glenn-jocher** [comment]
The hidden file is a temporary file generated during the optimization process and can be safely ignored or deleted after the export is complete.

**Pull request merged**